File: Dante1/Module1.06/Zadanie6.10/main.c
#include <stdio.h>

int counter_for_wiersz(int data[10][10], int j)
{
    int suma = 0;
    for(int i = 0; i < 10; i++)
    {
        suma = suma + data[i][j];
    }
    return suma;
}

int counter_for_column(int data[10][10], int i)
{
    int suma = 0;
    for(int j = 0; j < 10; j++)
    {
        suma = suma + data[i][j];
    }
    return suma;
}

int result(int data[10][10], int i, int j)
{
    int result = counter_for_wiersz(data,i) - counter_for_column(data,j);
    return -1 * result;
}

int main()
{
    int data[10][10];

    printf("Enter:\n");
    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
            int check = scanf("%i",&data[i][j]);
            if(check != 1)
            {
                printf("Incorrect input");
                return 1;
            }
        }
    }

    int pointer = 0;
    int max = data[0][0];

    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
            if(data[i][j] % 2 == 0)
            {
                if(data[i][j] > max)
                {
                    max = data[i][j];
                }
                pointer = 1;
            }
        }
    }
    if(pointer == 0)
    {
        printf("Nothing to show");
        return 0;
    }

    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
            if(max == data[i][j])
            {
                printf("%i %i %i\n",j,i,result(data,j,i));
            }
        }
    }

    return 0;
}

File: Dante1/Module1.06/Zadanie6.10/main.rs
fn counter_for_column(array: [[i32; 10]; 10], j: usize) -> i32
{
    let mut sum: i32 = 0;
    for i in 0..10
    {
        sum += array[i][j];
    }
    sum
}

fn counter_for_row(array: [[i32; 10]; 10], i: usize) -> i32
{
    let mut sum: i32 = 0;
    for j in 0..10
    {
        sum += array[i][j];
    }
    sum
}

fn result_counter(array: [[i32; 10]; 10], i: usize, j: usize) -> i32
{
    return -1 * (counter_for_column(array, j) - counter_for_row(array, i));
}

fn main()
{
    let mut array:[[i32; 10]; 10] = [[0; 10]; 10];
    let mut input: String = String::new();

    for i in 0..10
    {
        for j in 0..10
        {
            input.clear();

            std::io::stdin().read_line(&mut input).expect("Incorrect input");
            array[i][j] = input.trim().parse().expect("Incorrect parse");
        }
    }
    
    let mut pointer: bool = false;
    let mut max: i32 = array[0][0];
    
    for i in 0..10
    {
        for j in 0..10
        {
            if array[i][j] % 2 == 0
            {
                if array[i][j] > max
                {
                    max = array[i][j];    
                }
                pointer = true;
            }
        }
    }
    
    if pointer == false
    {
        panic!("Nothing to show");
    }
    
    for i in 0..10
    {
        for j in 0..10
        {
            if max == array[i][j]
            {
                println!("{} {} {}", i, j, result_counter(array, i, j));
            }
        }
    }
}
